### PR TITLE
Ensure PositionManager.moveLiquidity accrues interest

### LINKED ISSUE #690: Remaining dust with `positionManager.moveLiquidity` usage
When a user moves liquidity between a bucket with a staked or unstated NFT there is an accounting error causing some of the lps to stay in the `from` bucket and account for PositionManager.

image

Steps to replicate:
- create several lender positions with NFTs
- increase time
- use `PositionManager.moveLiqudity()`

### PR DESCRIPTION
**Pull request by EdNoepel**
# Description of change
## High level
Fixes issue #690 reported by Oasis team.
Implemented a test which fails upon `tearDown` without the fix.

# Description of bug or vulnerability and solution
This bug is caused because `PositionManager.moveLiquidity` is not accruing interest prior to calling `_lpsToQuoteToken`. As such, we ignore interest the lender earned since the last time pool interest had accrued, leaving unclaimable quote token in the bucket.

To remedy, `PositionManager.moveLiquidity` should call `Pool.updateInterest()` prior to `bucketInfo` and `_lpsToQuoteToken` calls.

# Contract size
Only `PositionManager` contract was touched.
## Pre Change
============ Deployment Bytecode Sizes ============
  PositionManager          -  18,688B  (76.04%)
## Post Change
============ Deployment Bytecode Sizes ============
  PositionManager          -  18,787B  (76.44%)

# Gas usage
Significant increase in cost of `moveLiquidity`, as expected.  Other gas changes perceived to be due to different test conditions.
## Pre Change
| src/PositionManager.sol:PositionManager contract |                 |        |        |         |         |
|--------------------------------------------------|-----------------|--------|--------|---------|---------|
| Function Name                                    | min             | avg    | median | max     | # calls |
| memorializePositions                             | 18112           | 320829 | 253448 | 1135007 | 69      |
| moveLiquidity                                    | 5864            | 189732 | 206150 | 466112  | 18      |
| reedemPositions                                  | 2879            | 35123  | 8039   | 140106  | 22      |
## Post Change
| src/PositionManager.sol:PositionManager contract |                 |        |        |         |         |
|--------------------------------------------------|-----------------|--------|--------|---------|---------|
| Function Name                                    | min             | avg    | median | max     | # calls |
| memorializePositions                             | 18112           | 315149 | 253448 | 1135007 | 62      |
| moveLiquidity                                    | 5864            | 237283 | 247632 | 614091  | 19      |
| reedemPositions                                  | 2879            | 37866  | 25538  | 140106  | 24      |

### CODE REVIEW

File: tests/forge/PositionManager.t.sol [hunk 1]
```diff
@@ -1844,6 +1844,142 @@ contract PositionManagerERC20PoolTest is PositionManagerERC20PoolHelperContract
...
+        });
+        _assertLenderLpBalance({
+            lender:      address(_positionManager),
+            index:       mintIndex,
+            lpBalance:   5_000 * 1e18,
+            depositTime: _startTime
+        });
+
+        // lender 1 moves liquidity
+        changePrank(address(lender1));
```
Comment (grandizzy) [inline tests/forge/PositionManager.t.sol]: we can remove cast of lender1 to address

**Review by grandizzy — APPROVED** [review]
Lgtm, minor comment. Thanks for fixing it!

File: tests/forge/PositionManager.t.sol (hunk 1 above)
Comment (EdNoepel) [reply tests/forge/PositionManager.t.sol]: done in 4480d2230f128b88f8acc3bf2b2420d9bd50db57

**Review by MikeHathaway — APPROVED** [review]
Change looks good - is there a plan for fixing the teardown issue?

**Comment by EdNoepel** [comment]
> Change looks good - is there a plan for fixing the teardown issue?

This PR introduces a new test, `testMoveLiquidityWithInterest`.  With these changes, tearDown of this test succeeds.  Without these changes, tearDown fails because the bucket has unclaimable quote token owned by `PositionManager`.

Unrelated, the rebase broke the tests, which I have just fixed in 78a63e3a4b7c7b7a942dafcd90faf27a0c45c683.

**Pull request merged**